includes/class-initializer.php

```diff
@@ -29,8 +29,11 @@ public static function init() {
 			Hub\Newspack_Ads_GAM::init();
 		}
 
-		if ( Site_Role::is_node() ) {
+		// Allow to access node settings before the site has a role, so it can be set via URL.
+		if ( Site_Role::is_node() || ! Site_Role::get() ) {
```

Comment:
With this change the "Node settings" menu is visible even before I define the site's role. Don't think it's ideal;

WDYT to process the incoming link in the "Site Role" settings page instead?

Reply:
Fixed in faab3894dd4e461164165ea7d6a054327355054b. Moving the incoming link processing to "Site Role" would not allow the code to leverage the existing settings-updating code.